server/http_handler_test.go

```diff
@@ -320,282 +313,268 @@ func (ts *HTTPHandlerTestSuite) regionContainsTable(c *C, regionID uint64, table
-	c.Assert(err, IsNil)
-	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, http.StatusOK)
-	c.Assert(binloginfo.SkippedCommitterCount(), Equals, int32(0))
+	require.NoError(t, err)
+	defer func() { require.NoError(t, resp.Body.Close()) }()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+	require.Equal(t, int32(0), binloginfo.SkippedCommitterCount())
 
 	binloginfo.EnableSkipBinlogFlag()
 	resp, err = ts.fetchStatus("/binlog/recover?op=nowait")
-	c.Assert(err, IsNil)
-	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, http.StatusOK)
-	c.Assert(binloginfo.IsBinlogSkipped(), Equals, false)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, resp.Body.Close()) }()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+	require.Equal(t, false, binloginfo.IsBinlogSkipped())
 
 	// Only the first should work.
 	binloginfo.EnableSkipBinlogFlag()
 	resp, err = ts.fetchStatus("/binlog/recover?op=nowait&op=reset")
-	c.Assert(err, IsNil)
-	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, http.StatusOK)
-	c.Assert(binloginfo.IsBinlogSkipped(), Equals, false)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, resp.Body.Close()) }()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+	require.Equal(t, false, binloginfo.IsBinlogSkipped())
 
 	resp, err = ts.fetchStatus("/binlog/recover?op=status")
-	c.Assert(err, IsNil)
-	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, http.StatusOK)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, resp.Body.Close()) }()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
 }
 
-func (ts *HTTPHandlerTestSuite) TestRegionsFromMeta(c *C) {
-	ts.startServer(c)
-	defer ts.stopServer(c)
-	resp, err := ts.fetchStatus("/regions/meta")
-	c.Assert(err, IsNil)
-	defer resp.Body.Close()
-	c.Assert(resp.StatusCode, Equals, http.StatusOK)
-
-	// Verify the resp body.
-	decoder := json.NewDecoder(resp.Body)
-	metas := make([]RegionMeta, 0)
-	err = decoder.Decode(&metas)
-	c.Assert(err, IsNil)
-	for _, meta := range metas {
-		c.Assert(meta.ID != 0, IsTrue)
-	}
-
-	// test no panic
-	c.Assert(failpoint.Enable("github.com/pingcap/tidb/server/errGetRegionByIDEmpty", `return(true)`), IsNil)
-	resp1, err := ts.fetchStatus("/regions/meta")
-	c.Assert(err, IsNil)
-	defer resp1.Body.Close()
-	c.Assert(failpoint.Disable("github.com/pingcap/tidb/server/errGetRegionByIDEmpty"), IsNil)
-}
-
-func (ts *basicHTTPHandlerTestSuite) startServer(c *C) {
+func (ts *basicHTTPHandlerTestSuite) startServer(t *testing.T) {
 	var err error
 	ts.store, err = mockstore.NewMockStore()
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	ts.domain, err = session.BootstrapSession(ts.store)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	ts.tidbdrv = NewTiDBDriver(ts.store)
 
 	cfg := newTestConfig()
 	cfg.Store = "tikv"
 	cfg.Port = 0
 	cfg.Status.StatusPort = 0
 	cfg.Status.ReportStatus = true
+	cfg.Socket = fmt.Sprintf("/tmp/%s.sock", t.Name())
```

Comment:
Is there any reason cfg.Socket is added in the server for tests? @tisonkun 

I guess it caused https://github.com/pingcap/tidb/issues/30642

Reply:
IIUC it's for using different socket file when running parallel tests.